lib/stat_tracker.rb

```diff
@@ -502,10 +524,55 @@ def favorite_opponent #issue # 29 - Fail due to not written
 
   end
 
-  def rival #issue # 30 - Fail due to not written
-
-
-
+  def rival_wins(team_id) #helper for #24 and possibly fave opp
+    rivals_wins = []
+    @games.each do |row|
+      if row[:away_team_id] == team_id.to_i
+        if row[:away_goals] < row[:home_goals]
+          rivals_wins << row[:home_team_id]
+        end
+      end
+      if row[:home_team_id] == team_id.to_i
+        if row[:home_goals] < row[:away_goals]
+          rivals_wins << row[:away_team_id]
+        end
+      end
+      # require "pry"; binding.pry 
+    end
+      rivals_wins_hash = rivals_wins.tally
```

Comment:
Interesting, I had not heard of the tally method. It's really cool! Thanks for introducing me to it.